api/api.go

```diff
@@ -43,15 +45,30 @@ func request(method string, u url.URL, body io.Reader) (resp *http.Response, err
 	req.Header.Set("User-Agent", ua)
 	req.Header.Add("Content-Type", "application/json")
 	req.Header.Add("Accept", "application/json")
+	if len(headers) == 1 {
```

Comment:
Should this be `>= 1` ? Seems a bit odd to use a `map[string][]string` if we are only expecting a single header value. Maybe we should accept `http.Header` as the arg instead?

Reply:
Because we're using a variadic function (and thus the caller could pass in as many `map[string][]string` arguments as they like), I've defined this header adding behavior to only run if there's a single collection of headers passed. 

That said, we could probably iterate over all the header arguments and merge them together. 

Will refactor this some more.

Reply:
As of da55706, the behavior is to merge all the headers passed into a single entity on the HTTP request.